I cleaned up the Makefile because the module runs fine without forcing off
signing or stack validation or setting any custom flags. This also changes the
recipe name so that the Gentoo ebuild default works.

I don't think upstream is interested in updating this code anyway. It was
intended to split out the GPU mod from the larger package "lights" for easier
testing but work has continued in that other package.

diff --git a/Makefile b/Makefile
index dd66372..dda9329 100644
--- a/Makefile
+++ b/Makefile
@@ -1,27 +1,11 @@
-CONFIG_MODULE_SIG=n
-CONFIG_STACK_VALIDATION=n
 MODULE_NAME = aura-gpu
 
-SRCS = \
-	asic/asic-polaris.c \
-	asic/asic-vega.c \
-	asic/asic-navi.c \
-	atom/atom.c \
-	aura-gpu-reg.c \
-	aura-gpu-i2c.c \
-	aura-gpu-bios.c \
-	aura-gpu-hw.c \
-	main.c
-
-KERNELDIR = /lib/modules/$(shell uname -r)/build
-PWD = $(shell pwd)
-KBUILD_EXTRA_SYMBOLS := $(ADAPTERDIR)/Module.symvers
-OBJS = $(SRCS:.c=.o)
-
 ifeq ($(KERNELRELEASE),)
 
-all:
-	$(MAKE) -C $(KERNELDIR) M=$(PWD) modules EXTRA_CFLAGS="-g -DDEBUG -I$(PWD)/../"
+KERNELDIR ?= /lib/modules/$(shell uname -r)/build
+PWD = $(shell pwd)
+
+all: module
 
 clean:
 	$(MAKE) -C $(KERNELDIR) M=$(PWD) clean
@@ -32,11 +16,26 @@ uninstall:
 install: uninstall all
 	sudo insmod $(MODULE_NAME).ko
 
-.PHONY: all clean uninstall install
+module:
+	$(MAKE) -C $(KERNELDIR) M=$(PWD) modules
+
+.PHONY: all clean uninstall install module
 
 else
 
-	obj-m += $(MODULE_NAME).o
-	$(MODULE_NAME)-y = $(OBJS)
+obj-m += $(MODULE_NAME).o
+
+$(MODULE_NAME)-y := \
+	asic/asic-polaris.o \
+	asic/asic-vega.o \
+	asic/asic-navi.o \
+	atom/atom.o \
+	aura-gpu-reg.o \
+	aura-gpu-i2c.o \
+	aura-gpu-bios.o \
+	aura-gpu-hw.o \
+	main.o
+
+ccflags-y := -DDEBUG
 
 endif
